src/main/java/com/code_intelligence/jazzer/junit/FuzzTestArgumentsProvider.java

```diff
@@ -155,42 +175,122 @@ private Stream<? extends Arguments> adaptInputsForFuzzTest(
+      FileSystem jar = FileSystems.newFileSystem(classInputsDirUri, new HashMap<>());
       // inputsDirUrl looks like this:
       // file:/tmp/testdata/ExampleFuzzTest_deploy.jar!/com/code_intelligence/jazzer/junit/testdata/ExampleFuzzTestInputs
-      String pathInJar = inputsDirUrl.getFile().substring(inputsDirUrl.getFile().indexOf('!') + 1);
-      return walkInputsInPath(jar.getPath(pathInJar)).onClose(() -> {
-        try {
-          jar.close();
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-      });
+      String pathInJar =
+          classInputsDirUrl.getFile().substring(classInputsDirUrl.getFile().indexOf('!') + 1);
+
+      Path classPathInJar = jar.getPath(pathInJar);
+
+      return Stream
+          .concat(walkClassInputs(classPathInJar), walkTestInputs(classPathInJar, testMethod))
+          .onClose(() -> {
+            try {
+              jar.close();
+            } catch (IOException e) {
+              throw new RuntimeException(e);
+            }
+          });
     } else {
-      throw new IOException("Unsupported protocol for inputs resource directory: " + inputsDirUrl);
+      throw new IOException(
+          "Unsupported protocol for inputs resource directory: " + classInputsDirUrl);
     }
   }
 
-  private static Stream<Map.Entry<String, byte[]>> walkInputsInPath(Path path) throws IOException {
+  /**
+   * Walks over the inputs for the method being tested, recurses into subdirectories
+   * @param classInputsPath the path of the class being tested, used as the base path where the test
+   *     method's directory
+   *                        should be
+   * @param testMethod the method being tested
+   * @return a stream of all files under {@code <classInputsPath>/<testMethod name>}
+   * @throws IOException can be thrown by the underlying call to {@link Files#find}
+   */
+  private static Stream<Map.Entry<String, byte[]>> walkTestInputs(
+      Path classInputsPath, Method testMethod) throws IOException {
+    Path testInputsPath = classInputsPath.resolve(testMethod.getName());
+    if (!Files.exists(testInputsPath)) {
+      return Stream.empty();
+    }
+    return walkInputsInPath(testInputsPath, Integer.MAX_VALUE);
+  }
+
+  /**
+   * Walks over the inputs for the class being tested. Does not recurse into subdirectories
+   * @param path the path to search to files
+   * @return a stream of all files (without directories) within {@code path}. If {@code path} is
+   *     {@code null}, then an
+   * empty stream is returned.
+   * @throws IOException can be thrown by the underlying call to {@link Files#find}
+   */
+  private static Stream<Map.Entry<String, byte[]>> walkClassInputs(Path path) throws IOException {
+    // this check is technically redundant thanks to the null check near the start of `walkInputs`
+    // however since these are only run once per tested method, I think it shouldn't degrade
+    // performance too much
```

Comment:
Instead of performing these checks, you could also try not checking and instead catching `FileNotFoundException`. Not that it would present any realistic problems here, but checking file existence separately from the access to the file could lead to unexpected behavior when the file is deleted between the two operations.